box: apply dynamic cfg even if option value is unchanged (#8552)
tarantool/tarantool: changelogs/unreleased/gh-8551-box-cfg-reload-fix.md

## bugfix/core

>  * `box.cfg.listen` and `box.cfg.replication` are now reconfigured even if
>    the URI is unchanged (gh-8551).

Reviewer: ```suggestion
 * `box.cfg.listen` and `box.cfg.replication` can now be reconfigured
   without changing the URI (gh-8551).
```

Author: Applied the suggestion. Thanks.
